src/Nethermind/Nethermind.Merge.Plugin/Handlers/V1/NewPayloadV1Handler.cs

```diff
@@ -251,20 +261,14 @@ private PayloadStatusV1 BuildExecutePayloadResult(ExecutionPayloadV1 request, bo
             {
                 payloadStatus.ValidationError = validationMessage;
                 payloadStatus.Status = PayloadStatus.Invalid;
-                if (_lastValidHashes.ContainsKey(request.ParentHash))
+                if (_invalidChainTracker.IsOnKnownInvalidChain(request.BlockHash, out Keccak? lastValidHash))
```

Comment:
Shouldn't we remove BuildExecutePayloadResult now?

Reply:
Good point. I thought something else is in there.

Reply:
On second thought, it is also used on line 178 also and I don't want to mess around it right now.